### PR TITLE
Appcallback versioning

### PR DESCRIPTION
**Pull request by vinayada1**
# Description

Add Appcallback versioning for HTTP APIs

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #1678 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ *] Code compiles correctly
* [ *] Created/updated tests
* [ ] Unit tests passing
* [ *] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ *] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: tests/apps/service_invocation/app.go [hunk 1]
```diff
@@ -216,16 +216,16 @@ func appRouter() *mux.Router {
...
+	router.HandleFunc("/v1.0/multihop", multihopHandler).Methods("POST")
 
 	router.HandleFunc("/tests/invoke_test", testHandler)
 
 	// these are called through dapr service invocation
-	router.HandleFunc("/posthandler", postHandler).Methods("POST")
-	router.HandleFunc("/gethandler", getHandler).Methods("GET")
-	router.HandleFunc("/puthandler", putHandler).Methods("PUT")
-	router.HandleFunc("/deletehandler", deleteHandler).Methods("DELETE")
+	router.HandleFunc("/v1.0/posthandler", postHandler).Methods("POST")
```
Comment (youngbupark) [inline tests/apps/service_invocation/app.go:225]: I would define const for v1.0 e.g. AppCallbackV1 = "v1.0"

File: pkg/channel/http/http_channel.go [hunk 2]
```diff
@@ -121,8 +121,8 @@ func (h *Channel) invokeMethodV1(ctx context.Context, req *invokev1.InvokeMethod
 func (h *Channel) constructRequest(ctx context.Context, req *invokev1.InvokeMethodRequest) *fasthttp.Request {
 	var channelReq = fasthttp.AcquireRequest()
 
-	// Construct app channel URI: VERB http://localhost:3000/method?query1=value1
-	uri := fmt.Sprintf("%s/%s", h.baseAddress, req.Message().GetMethod())
+	// Construct app channel URI: VERB http://localhost:3000/v1.0/method?query1=value1
+	uri := fmt.Sprintf("%s/v1.0/%s", h.baseAddress, req.Message().GetMethod())
```
Comment (youngbupark) [inline pkg/channel/http/http_channel.go:125]: Can you please define const for v1.0 ?

**Comment by amanbha** [comment]
app callback versioning is not needed for topics, binding and method invocation.

Also we need clarity that when 2.0 is implemented, what will be called at startup (app is still implementing c1.0):
v1.0/dapr/config or v2.0/dapr/config
v1.0/dapr/subscribe or v2.0/dapr/subscribe


Also for actors callback, once v2.0 is implemeted (app is still implementing c1.0):
How would runtime determine which version to call?

Could you please set up sometime to discuss it with @yaron2 @youngbupark and me

**Comment by vinayada1** [comment]
We decided to not implement appcallback versioning for HTTP. Changes no longer needed

**Comment by artursouza** [comment]
Ack.

**Pull request closed without merging**